src/ec2_metadata/__init__.py

```diff
@@ -107,6 +107,13 @@ def iam_info(self) -> dict[str, Any] | None:
     def instance_action(self) -> str:
         return self._get_url(f"{self.metadata_url}instance-action").text
 
+    @property
+    def spot_instance_action(self) -> dict[str, Any] | None:
```

Comment:
move attribute to be in alphabetical order

Reply:
The docs imply this is `dict[str, str]`

Also, I think it'd be better if we parsed the contents for users into an object, since otherwise that will just require users to parse the datetime string.